code/modules/mob/living/silicon/ai/say.dm

```diff
@@ -134,6 +134,22 @@ var/const/VOX_PATH = "sound/vox_fem/"
 	for(var/word in words)
 		play_vox_word(word, src.z, null)
 
+	ai_voice_announcement_to_text(words, src)
+
+
+/proc/ai_voice_announcement_to_text(words, ai)
+	var/mob/living/silicon/ai/AI = ai
+	var/words_string = jointext(words, " ")
+	var/formatted_message = "<h1 class='alert'>A.I. Announcement</h1>"
+	formatted_message += "<br><span class='alert'>[words_string]</span>"
+	formatted_message += "<br><span class='alert'> -[AI]</span>"
+
+	for(var/mob/M in GLOB.player_list)
+		if(M.client && !(M.client.prefs.sound & SOUND_AI_VOICE))
+			var/turf/T = get_turf(M)
+			if(T && T.z == AI.z && M.can_hear())
+				M << 'sound/misc/notice2.ogg'
+				to_chat(M, formatted_message)

```

Comment:
Instead of creating a new announcement proc you should just use the existing proc and send the words to that

Reply:
oh but then you have the issue of people who CAN hear. I see. Maybe a var?

Reply:
oh you could just throw it in the for loop

Reply:
Throw what in the for loop?

Reply:
the announcement proc but it probably doesnt work that way so nevermind